### DISCUSSION TITLE
Connect wallet failed

### DISCUSSION THREAD

**Discussion by ffej8797** [WalletConnect/walletconnect-monorepo#3670] [Developer Support]
Hello, I have some problems about connect wallets. 

1. I connected Metamask successfully at first time. Then I logged out and connect again. It popup the connect message and I click connect. But it will stuck in loading. Then I close Metamask and connect again, it works. How can I make it connect successfully without close Metamask. 
Here is the video about the situation: [](https://drive.google.com/file/d/1MzkdSpmLLojGAGwmToetUpfBIs1U5I3G/view?usp=sharing)

2. I connected Trust Wallet successful. Then I disconnected. And I want to connected Uniswap wallet but it failed. It didn't popup the connect button let me connect.

How can I fix these problems?

**Comment by glitch-txs** [comment]
Hi, could you share a [MRE](https://github.com/orgs/WalletConnect/discussions/3457) of this issue? Thanks! :)

**Comment by ffej8797** [comment]
I use the demo on here: https://app.web3inbox.com/login?next=%2Fnotifications
The situation is same ion this video: https://drive.google.com/file/d/1UYToLgk6wl_hxo7qbyKbrQYR9pSLwLwe/view?usp=drive_link

**Reply by glitch-txs** [reply]
Thanks for sharing! Could you test this on labs? https://lab.web3modal.com/

**Comment by Rossiross** [comment]
I have the same problem, connect to the wallet.

**Reply by Schellpr** [reply]
It happens that I get out of my wallet, I go into my wallet three times.

**Reply by SamsonchikSnayder** [reply]
Yes, this happens. To be honest, it strains this kind of movement.

**Comment by KononWW** [comment]
Perhaps Metamask is buggy?

**Comment by Pretty7up** [comment]
the wallet may not connect for many reasons. Try reinstalling the application first. If it doesn’t help, we’ll figure it out)
